ogr/ogrsf_frmts/mem/ogrmemlayer.cpp

```diff
@@ -819,6 +820,74 @@ OGRErr OGRMemLayer::AlterFieldDefn( int iField, OGRFieldDefn *poNewFieldDefn,
     return OGRERR_NONE;
 }
 
+/************************************************************************/
+/*                         AlterGeomFieldDefn()                         */
+/************************************************************************/
+
+OGRErr OGRMemLayer::AlterGeomFieldDefn( int iGeomField,
+                                        const OGRGeomFieldDefn* poNewGeomFieldDefn,
+                                        int nFlagsIn )
+{
+    if( !m_bUpdatable )
+        return OGRERR_FAILURE;
+
+    if( iGeomField < 0 || iGeomField >= m_poFeatureDefn->GetGeomFieldCount() )
+    {
+        CPLError(CE_Failure, CPLE_NotSupported, "Invalid field index");
+        return OGRERR_FAILURE;
+    }
+
+    auto poFieldDefn = m_poFeatureDefn->GetGeomFieldDefn(iGeomField);
+
+    if( nFlagsIn & ALTER_GEOM_FIELD_DEFN_NAME_FLAG )
+        poFieldDefn->SetName(poNewGeomFieldDefn->GetNameRef());
+    if( nFlagsIn & ALTER_GEOM_FIELD_DEFN_TYPE_FLAG )
+    {
+        if( poNewGeomFieldDefn->GetType() == wkbNone )
+            return OGRERR_FAILURE;
+        poFieldDefn->SetType(poNewGeomFieldDefn->GetType());
```

Comment:
I'm guessing it's purely operator error if there's existing features with incompatible geometries compared to the new type, right?

Reply:
Ah, just saw the dox note below -- :+1: